cmd/agent/daemon/enrichment/enrichers.go

```diff
@@ -63,37 +68,45 @@ func (enricher *fileHashEnricher) Enrich(ctx context.Context, req *EnrichedConta
 		return
 	}
 
+	metricLabels := metricLabelsFromContainer(req.EbpfEvent.Container)
+	enrichMetric := metrics.AgentEnricherEventsTotalEnriched.WithLabelValues(metricLabels...)
+	enrichProcMissingMetric := metrics.AgentFileHashEnricherProcMissingTotal.WithLabelValues(metricLabels[1:]...)
+	enrichErrorMetric := metrics.AgentEnricherEventsTotalErrors.WithLabelValues(metricLabels...)
+
 	sha, err := enricher.calcFileHashForPID(req.EbpfEvent.Container, proc.PID(req.EbpfEvent.Context.NodeHostPid), exec.Path)
-	if err != nil {
-		if errors.Is(err, ErrFileDoesNotExist) {
-			return
-		}
-	} else {
+	if err == nil {
 		setExecFileHash(exec, sha)
+		enrichMetric.Inc()
 		return
 	}
+	enrichProcMissingMetric.Inc()
 
+	// If we can't find the file hash from the event PID, we try to find it from other PIDs in the same mount namespace
 	for _, pid := range enricher.mountNamespacePIDStore.GetBucket(proc.NamespaceID(req.EbpfEvent.Context.MntID)) {
 		if pid == proc.PID(req.EbpfEvent.Context.NodeHostPid) {
 			// We already tried that PID of the event, skipping.
 			continue
 		}
 
 		sha, err := enricher.calcFileHashForPID(req.EbpfEvent.Container, pid, exec.Path)
```

Comment:
I just notice we do not cleanup PIDs from the buckets, which can cause issues, since PIDs will get reused. We should fix this in a follow up PR.

Reply:
Good catch, I will open a different PR to handle this case